src/test/compile-fail/rfc-2008-non-exhaustive/auxiliary/enums.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2012 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![crate_type = "rlib"]
+#![feature(non_exhaustive)]
+
+#[non_exhaustive]
+pub enum NormalEnum {
+    FirstVariant,
+    SecondVariant
+}
+
+pub enum FieldEnum {
+    WithoutFields,
+    #[non_exhaustive] WithAnonStruct { field: u32 },
+    #[non_exhaustive] WithTuple(u32)
```

Comment:
I'd like to see all of these configurations tested:
```
#[non_exhaustive]
enum NonExhaustiveEnum {
    Unit,
    Tuple(u8),
    Struct { field: u8 },
}

enum NonExhaustiveVariants {
    #[non_exhaustive] Unit,
    #[non_exhaustive] Tuple(u8),
    #[non_exhaustive] Struct { field: u8 },
}
```

Reply:
Now testing with these enums.